src/arr/compiler/desugar.arr

```diff
@@ -357,6 +357,8 @@ fun desugar-expr(expr :: A.Expr):
       A.s-module(l, desugar-expr(answer), dv, dt, desugar-expr(provides), types.map(desugar-afield), desugar-expr(checks))
     | s-instantiate(l, inner-expr, params) =>
       A.s-instantiate(l, desugar-expr(inner-expr), params.map(desugar-ann))
+    | s-hint-exp(l, hints, exp) =>
```

Comment:
I believe we should go with the other way: remove `s-hint-exp` from Pyret. Note that these are all occurrences of it in the repo:

```
./src/arr/trove/ast-visitors.arr:    method s-hint-exp(self, l, hints, exp):
./src/arr/trove/ast-visitors.arr:    method s-hint-exp(self, l, hints, exp):
./src/arr/trove/ast-visitors.arr:      s-hint-exp(l, self.list(hints), exp.visit(self))
./src/arr/trove/ast-visitors.arr:    method s-hint-exp(self, l, hints, exp):
./src/arr/trove/ast-visitors.arr:      s-hint-exp(dummy-loc, self.list(hints), exp.visit(self))
./src/arr/trove/ast.arr:  | s-hint-exp(l :: Loc, hints :: List<Hint>, exp :: Expr) with:
./src/arr/trove/ast.arr:    method label(self): "s-hint-exp" end,
./src/arr/trove/ast.arr:  method s-hint-exp(self, l :: Loc, hints :: List<Hint>, exp :: Expr):
./src/arr/trove/ast.arr:    s-hint-exp(l, hints, exp.visit(self))
./src/arr/trove/ast.arr:  method s-hint-exp(self, l :: Loc, hints :: List<Hint>, exp :: Expr):
./src/arr/trove/ast.arr:  method s-hint-exp(self, l :: Loc, hints :: List<Hint>, exp :: Expr):
./src/arr/trove/ast.arr:    s-hint-exp(dummy-loc, hints, exp.visit(self))
./src/arr/desugar/conversion-visitor.arr:    method s-hint-exp(self, l, hints, exp):
./src/arr/desugar/conversion-visitor.arr:      g-surf("s-hint-exp",
./src/arr/desugar/conversion-visitor.arr:    | op == "s-hint-exp"
./src/arr/desugar/conversion-visitor.arr:      s-hint-exp(maybe-loc.value,
./src/arr/desugar/stepify.arr:      A.s-program(l, prov, provt, imp, A.s-hint-exp(l, [list: A.h-trace], body))
./src/arr/desugar/stepify.arr:  #     method s-hint-exp(self, l, hints, expr):
./src/arr/desugar/stepify.arr:  #         A.s-hint-exp(l, hints, expr.visit(self))
./src/arr/desugar/quote.arr:  # "s-hint-exp", A.s-hint-exp,
./src/arr/compiler/type-check.arr:        | s-hint-exp(l, hints, exp) =>
./src/arr/compiler/type-check.arr:          raise("checking for s-hint-exp not implemented")
./src/arr/compiler/type-check.arr:    | s-hint-exp(l, hints, exp) =>
./src/arr/compiler/type-check.arr:      raise("synthesis for s-hint-exp not implemented")
./src/arr/compiler/well-formed.arr:  method s-hint-exp(_, l :: Loc, hints :: List<A.Hint>, exp :: A.Expr):
./src/arr/compiler/well-formed.arr:    well-formed-visitor.s-hint-exp(l, hints, exp)
./src/arr/compiler/desugar.arr:    | s-hint-exp(l, hints, exp) =>
./src/arr/compiler/desugar.arr:      A.s-hint-exp(l, hints, desugar-expr(exp))
```

showing that it's not used in a meaningful way at all. And the node couldn't be constructed by the parser.

@jpolitz @blerner  thoughts?